### ISSUE TITLE
[Bug] Possible memory problem on TripsLayer

### ISSUE THREAD

**Issue by ponytojas** [visgl/deck.gl#8139]
### Description

**TL;DR**: It looks to me that update frequently a TripsLayer make it accumulate data and making a tab to crash.

------------------------------------------------------------------------------------------------------------------------------

I'm currently working on a Public Transport Map based app.
One of the features is related to be able to display the current position in the map of the different vehicles.

To be able to do that I preprocess on my backend the TripsLayer Data getting an array with the following structure

```
[{
id: "SomeTripId",
path: [[X,Y], [X2,Y2]...],
timestamps: [MsFromMidnight, MsFromMidnight2, ...],
properties: {color: [X, Y, Z]}
}, ...]
```

I calculate the following 45 minutes of paths and timestamps, resulting in between 15 and 40 elements per value for each one.
This calculation is done in the backend every 30 seconds and requested by the frontend every 45.

Once I have a new data I use the following code to create a new layer and replace the old one as suggested [here](https://deck.gl/docs/faq#should-i-really-regenerate-all-layers-every-time-something-changes)
```jsx
  useEffect(() => {
    if (!vehiclesTrips) return;
    if (vehiclesTrips?.length === 0) return;

    if (tilesModeRef.current !== tilesMode) {
      setLayers((prev) => prev.filter((l) => l.id !== TRIPS_LAYER_CONFIG.options.id));
      tilesModeRef.current = tilesMode;
    }

    const widthPixel = zoom > 14 ? 4 : 2;
    const visible = (zoom > 10) && enabledLayers[LAYERS_IDS.TRIP_LAYER];

    TRIPS_LAYER_CONFIG.options.visible = visible;
    TRIPS_LAYER_CONFIG.options.widthMinPixels = widthPixel;
    TRIPS_LAYER_CONFIG.options.currentTime = time;
    TRIPS_LAYER_CONFIG.options.onHover = (info) => onHover(onHoverTrip(info));
    TRIPS_LAYER_CONFIG.options.onClick = (info) => onTripClick(info);

    const tripsLayer = new TripsLayer({ ...TRIPS_LAYER_CONFIG.options, data: vehicleTripsRef.current });
    setLayers((prev) => [...prev.filter((l) => l.id !== TRIPS_LAYER_CONFIG.options.id), tripsLayer]);
  }, [enabledLayers, time, zoom, vehiclesTrips, tilesMode]);
```

Where TRIPS_LAYER_CONFIG is the following
```json
{
  "type": "trips",
  "options": {
    "getPath": (d) => (d.path),
    "getTimestamps": (d) => (d.timestamps),
    "getColor": (d) => d.properties.color,
    "pickable": true,
    "opacity": 1,
    "trailLength": 45000,
    "getWidth": 4,
    "jointRounded": true,
    "widthMinPixels": 4,
    "widthScale": 1,
    "widthUnits": "meters",
    "capRounded": true,
    "id": "trips",
    "visible": false,
    "currentTime": 35170539
  }
}
```

Everything works correctly during some time but then the app crash with an `Out of memory` error.

Looking at the Memory tab in the Developer tools of the browser I can see how the memory usage is increased every time the app gets the updated data from the backend and is passed to the new layer.

This causes the application to crashafter 15 to 30 minutes.

For me looks like the TripsLayer is not correctly cleaning old data and accumulating it.

### Flavors

- [ ] Script tag
- [X] React
- [ ] Python/Jupyter notebook
- [ ] MapboxOverlay
- [ ] GoogleMapsOverlay
- [ ] CartoLayer
- [ ] ArcGIS

### Steps to Reproduce

[Link for a sample dataset](https://drive.google.com/file/d/1daR9tA_bLAtUQcwfN633qpCNByQJBAkF/view?usp=sharing)

### Environment

- Framework version: 8.9.30
- Browser: Edge / Chrome
- OS: Windows 11

**Comment by Pessimistress** [comment]
You should check the number of trips and the number of points per trip as time accumulates. The layer itself does not keep old data around, but somewhere else in your React component might. Chrome's memory profiler can give you a better look into what is consuming so much.

Closing this issue because there is no evidence this is a bug in deck.gl. Feel free to open a thread in Discussions.
